docs: allow for errors to be thrown and shown in documentation (#324)
limebit/medmodels: docs/user_guide/02b_query_engine.md

:::

:::{note}
Query methods used for changing the operands cannot be assigned to variables, for further querying, since their _Return_ is `None`. That is, the following code snippet will set `gender_lowercase` as None, and as a result, an `AttributeError` will be thrown:

```{exec_error-literalinclude} scripts/02b_query_engine.py
---
language: python
setup-lines: 1-6
lines: 61-68

> ---

Reviewer: it should be configurable which type of error is expected here. In this case it would be a `PanicException`.
To fulfill the issue requirement
> A distinction between accidental errors and intended errors needs to be made

:::

:::{note}

> Query methods used for changing the operands cannot be assigned to variables, for further querying, since their _Return_ is `None`. That is, the following code snippet will set `gender_lowercase` as None, and as a result, an `AttributeError` will be thrown:

Reviewer: check the commas here. I think there might be too many

> Query methods used for changing the operands cannot be assigned to variables, for further querying, since their _Return_ is `None`. That is, the following code snippet will set `gender_lowercase` as None, and as a result, an `AttributeError` will be thrown:

Reviewer: ```suggestion
Query methods used for changing the operands cannot be assigned to variables, for further querying, since their return type is `None`. That is, the following code snippet will set `gender_lowercase` as None, and as a result, an `AttributeError` will be thrown:
```

Reviewer: The `that is` sounds a bit weird at the beginning. You didn't change it but you're already here

> Query methods used for changing the operands cannot be assigned to variables for further querying, since their return type is `None`. That can be appreciated in the following code snippet, where the variable `gender_lowercase` in the query takes None as value. An `AttributeError` is thrown as a consequence when trying to further query with the `equal_to` querying method:

Reviewer: ```suggestion
Query methods used for changing the operands cannot be assigned to variables for further querying, since their return type is `None`. The following code snippet shows an example, where the variable `gender_lowercase` evaluates to None. An `AttributeError` is thrown as a consequence when trying to further query with the `equal_to` querying method:
```
